indexing-hadoop/src/test/java/io/druid/indexer/HdfsClasspathSetupTest.java

```diff
@@ -0,0 +1,192 @@
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.indexer;
+
+import com.metamx.common.StringUtils;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import junit.framework.Assert;
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.DistributedFileSystem;
+import org.apache.hadoop.hdfs.MiniDFSCluster;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.MRJobConfig;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class HdfsClasspathSetupTest
+{
+  private static MiniDFSCluster miniCluster;
+  private static File hdfsTmpDir;
+  private static Configuration conf;
+  public static Path finalClasspath = new Path("/tmp/classpath");
+  public static Path intermediatePath = new Path("/tmp/intermediate");
+  private static String dummyJarString = "This is a test jar file.";
+  private File dummyJarFile;
+
+  @BeforeClass
+  public static void setupStatic() throws IOException, ClassNotFoundException
+  {
+    hdfsTmpDir = File.createTempFile("hdfsClasspathSetupTest", "dir");
+    hdfsTmpDir.deleteOnExit();
+    if (!hdfsTmpDir.delete()) {
+      throw new IOException(String.format("Unable to delete hdfsTmpDir [%s]", hdfsTmpDir.getAbsolutePath()));
+    }
+    conf = new Configuration(true);
+    conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, hdfsTmpDir.getAbsolutePath());
+    miniCluster = new MiniDFSCluster.Builder(conf).build();
+  }
+
+  @Before
+  public void setUp() throws IOException
+  {
+    dummyJarFile = File.createTempFile("dummy-test", ".jar");
+    dummyJarFile.deleteOnExit();
+    Files.copy(
+        new ByteArrayInputStream(StringUtils.toUtf8(dummyJarString)),
+        dummyJarFile.toPath(),
+        StandardCopyOption.REPLACE_EXISTING
+    );
+  }
+
+  @AfterClass
+  public static void tearDownStatic() throws IOException
+  {
+    if (miniCluster != null) {
+      miniCluster.shutdown(true);
+    }
+  }
+
+  @After
+  public void tearDown() throws IOException
```

Comment:
Can you add checks here to make sure the cleanup works as intended. Like make sure the files do not exist after the deletes are called?